ide/che-core-ide-stacks/src/main/resources/stacks.json

```diff
@@ -163,6 +163,146 @@
       "mediaType": "image/svg+xml"
     }
   },
+  {
+    "id": "java-mysql-theia-openshift",
+    "creator": "ide",
+    "name": "Che 7 - Java + MySQL",
+    "description": "Multi Container Workspace -Java + MySQL in Theia IDE",
+    "scope": "general",
+    "tags": [
+      "Java 1.8",
+      "Theia",
+      "MySQL"
+    ],
+    "workspaceConfig": {
+      "environments": {
+        "default": {
+          "recipe": {
+            "contentType": "text/x-yaml",
+            "type": "openshift",
+            "content": "kind: List\nitems:\n - \n  apiVersion: v1\n  kind: Pod\n  metadata:\n   name: web\n   labels:\n    type: db\n  spec:\n   containers:\n    - \n     image: eclipse/centos_jdk8\n     name: dev\n     ports:\n      - \n       containerPort: 8080\n       protocol: TCP\n     resources:\n      limits:\n       memory: 512Mi\n    - \n     image: centos/mysql-57-centos7\n     name: mysql\n     ports:\n      - \n       name: mysql\n       containerPort: 3306\n       protocol: TCP\n     env:\n      - \n       name: MYSQL_USER\n       value: petclinic\n      - \n       name: MYSQL_ROOT_PASSWORD\n       value: password\n      - \n       name: MYSQL_PASSWORD\n       value: password\n      - \n       name: MYSQL_DATABASE\n       value: petclinic\n     resources:\n      limits:\n       memory: 256Mi\n - \n  apiVersion: v1\n  kind: Service\n  metadata:\n   name: db\n  spec:\n   selector:\n    type: db\n   ports:\n    - \n     name: mysql\n     port: 3306\n     protocol: TCP\n     targetPort: mysql\n"
```

Comment:
Stack description says nothing about `petclinic`, maybe it would be better to set some general values for Database env vars? Like `MYSQL_USER=user`, `MYSQL_PASSWORD=password` =)

Reply:
Also, you specified memory limit in the recipe and in machine configuration for `web/dev` machine. I think it would be better to choose one place for configuring memory.

Reply:
@sleshchenko what is preferable place to configure memory? I found it quite confusing with both places.

Reply:
> @sleshchenko what is preferable place to configure memory?

It's hard to say what is preferable place in this particular case. A recipe is a place where a user defines his runtime (ideal case - the same recipe as he uses for running application on production) and he may define needed limits there.
Che gives an ability to override memory limit in a recipe by specifying a corresponding value in machine configuration.
Up to you which place you choose.

> I found it quite confusing with both places.

It's why I asked you to choose one place =) Because it is confusing that you configure and then override memory limit.

Reply:
As about user that's should be arranges for specific project. Here we have command which clones petclinic sample. But it is easy to change the value in dashboard.

Reply:
@mmorhun 

> As about user that's should be arranges for specific project. Here we have command which clones petclinic sample. But it is easy to change the value in dashboard.

Sorry, I missed this command before. 
Well, I think stack's purpose is more than just petclinic application and it makes more sense to configure petclinic app (like with `sed` command after cloning) instead of configuring MYSQL container for petclinic sample. But up to you change or not